pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,627 @@
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/lws-operator/bindata"
+	leaderworkersetapiv1 "github.com/openshift/lws-operator/pkg/apis/leaderworkersetoperator/v1"
+	leaderworkersetoperatorv1clientset "github.com/openshift/lws-operator/pkg/generated/clientset/versioned/typed/leaderworkersetoperator/v1"
+	operatorclientinformers "github.com/openshift/lws-operator/pkg/generated/informers/externalversions/leaderworkersetoperator/v1"
+	"github.com/openshift/lws-operator/pkg/operator/operatorclient"
+)
+
+type TargetConfigReconciler struct {
+	ctx                           context.Context
+	targetImage                   string
+	operatorClient                leaderworkersetoperatorv1clientset.LeaderWorkerSetOperatorInterface
+	dynamicClient                 dynamic.Interface
+	leaderWorkerSetOperatorClient *operatorclient.LeaderWorkerSetClient
+	kubeClient                    kubernetes.Interface
+	apiextensionClient            *apiextclientv1.Clientset
+	eventRecorder                 events.Recorder
+	queue                         workqueue.TypedRateLimitingInterface[string]
+	namespace                     string
+}
+
+func NewTargetConfigReconciler(
+	ctx context.Context,
+	targetImage string,
+	namespace string,
+	operatorConfigClient leaderworkersetoperatorv1clientset.LeaderWorkerSetOperatorInterface,
+	operatorClientInformer operatorclientinformers.LeaderWorkerSetOperatorInformer,
+	leaderWorkerSetOperatorClient *operatorclient.LeaderWorkerSetClient,
+	dynamicClient dynamic.Interface,
+	kubeClient kubernetes.Interface,
+	apiExtensionClient *apiextclientv1.Clientset,
+	eventRecorder events.Recorder,
+) *TargetConfigReconciler {
+	c := &TargetConfigReconciler{
+		ctx:                           ctx,
+		operatorClient:                operatorConfigClient,
+		dynamicClient:                 dynamicClient,
+		leaderWorkerSetOperatorClient: leaderWorkerSetOperatorClient,
+		kubeClient:                    kubeClient,
+		apiextensionClient:            apiExtensionClient,
+		eventRecorder:                 eventRecorder,
+		queue:                         workqueue.NewTypedRateLimitingQueueWithConfig(workqueue.DefaultTypedControllerRateLimiter[string](), workqueue.TypedRateLimitingQueueConfig[string]{Name: "TargetConfigReconciler"}),
+		targetImage:                   targetImage,
+		namespace:                     namespace,
+	}
+
+	operatorClientInformer.Informer().AddEventHandler(c.eventHandler())
```

Comment:
We can use a controller factory from library-go here

Reply:
That is great instead of hacky looking code portion. Used.